src/unlabel.ml

```diff
@@ -0,0 +1,192 @@
+        labels;
+      Format.pp_close_box fmt ();
+      Format.pp_close_box fmt ();
+      Format.pp_print_cut fmt ();
+    end
+    labelled_functions;
+  *)
+
+  (* Create list containing only labels and postions, but no function names. *)
+  let labels =
+    List.map (fun (_,labels,_) -> labels) labelled_functions
+    |> List.flatten
+  in
+
+  begin (* check whether positions are in order *)
+    List.map (fun (_, {Lexing.pos_cnum}) -> pos_cnum) labels
+    |> List.fold_left (fun a b -> assert (a <= b); b) 0
+    |> ignore
+  end;
+
+  Format.pp_open_vbox fmt 0;
+  Format.pp_open_vbox fmt 0;
+  Format.pp_open_vbox fmt 0;
+  begin (* strip labels from mli *)
+    let unlabelled_mli = Buffer.create (String.length mli_text) in
+    let open String in
+    let rec strip_labels start = function
+      | (label, {Lexing.pos_cnum = pos}) :: tail ->
+        assert (label = sub mli_text pos (length label));
+        Buffer.add_substring unlabelled_mli mli_text start (pos-start);
+        let colon = index_from mli_text pos ':' in
+        strip_labels (colon+1) tail
+      | [] -> start
+    in
+    let rec unlabel start = function
+      | [] ->
+        Buffer.add_substring unlabelled_mli mli_text
+          start (String.length mli_text - start)
+      | (name, labels, _) :: tail ->
+        let start = strip_labels start labels in
+        let start = try
+            let ds_start = Str.(search_forward (regexp "(\\*\\*") mli_text start) in
+            let ds_end = Str.(search_forward (regexp "\\*)") mli_text ds_start) + 2 in
+            if
+              match tail with
+              | [] -> true (* last declaration - we may process until EOF. *)
+              | (_, _, { Location.loc_start = { pos_cnum = next_decl }}) :: _
+                when ds_end - 1 <= next_decl
+                -> true
+              | _ -> false
+            then begin
+              Buffer.add_substring unlabelled_mli mli_text
+                start (ds_start - start);
+              let docstring = sub mli_text ds_start (ds_end - ds_start) in
+              let stripped_docstring =
+                List.fold_left
+                  begin fun docstring (label, _) ->
+                    Str.global_replace
+                      (Str.regexp_string ("~" ^ label)) label
+                      docstring
+                  end
+                  docstring
+                  labels
+              in
+              Buffer.add_string unlabelled_mli stripped_docstring;
+              ds_end
+            end
+            else raise Not_found
+          with Not_found -> start
+        in
+        unlabel start tail
+    in unlabel 0 labelled_functions;
+
+    (* hacks *)
+    let unlabelled_mli =
+      Buffer.to_bytes unlabelled_mli |> Bytes.unsafe_to_string
+      |> Str.(global_replace (regexp_string labelled_name) unlabelled_name )
+    in
+
+    let out = open_out unlabelled_mli_filename in
+    output_string out (
+      "(*\n *\n *\n *      AUTOGENERATED FROM " ^
```

Comment:
this output is a bit verbose for my taste, a single-line comment should be enough

Reply:
changed this to a single line comment.